File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.go
func matchPlayersAndTrainers(players []int, trainers []int) int {
    sort.Slice(players, func (i,j int) bool { return players[i] < players[j] })
    sort.Slice(trainers, func (i,j int) bool { return trainers[i] < trainers[j] })

    j := 0
    matches := 0

    for _, ability := range players {
        for j < len(trainers) && ability > trainers[j] {
            j++
        }

        if j == len(trainers) {
            return matches
        }

        matches++
        j++
    }

    return matches
}

File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort(); trainers.sort()

        matches = 0

        j = 0

        for player in players:
            while j < len(trainers) and trainers[j] < player:
                j += 1
            
            if j == len(trainers):
                return matches

            matches += 1

            j += 1
        
        return matches
